File: app/build-app2/tmp/kapt3/stubs/debug/eu/villacristina/lechugapro/ui/comercializacion/cliente/ClienteListaViewModel.java
package eu.villacristina.lechugapro.ui.comercializacion.cliente;

@kotlin.Metadata(mv = {1, 9, 0}, k = 1, xi = 48, d1 = {"\u0000<\n\u0002\u0018\u0002\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0002\b\u0002\n\u0002\u0018\u0002\n\u0002\u0010\"\n\u0002\u0010\t\n\u0002\b\u0003\n\u0002\u0010 \n\u0002\u0018\u0002\n\u0002\b\u0002\n\u0002\u0010\u0006\n\u0002\b\u0002\u0018\u00002\u00020\u0001B\u0015\u0012\u0006\u0010\u0002\u001a\u00020\u0003\u0012\u0006\u0010\u0004\u001a\u00020\u0005\u00a2\u0006\u0002\u0010\u0006R\u001d\u0010\u0007\u001a\u000e\u0012\n\u0012\b\u0012\u0004\u0012\u00020\n0\t0\b\u00a2\u0006\b\n\u0000\u001a\u0004\b\u000b\u0010\fR\u001d\u0010\r\u001a\u000e\u0012\n\u0012\b\u0012\u0004\u0012\u00020\u000f0\u000e0\b\u00a2\u0006\b\n\u0000\u001a\u0004\b\u0010\u0010\fR\u0017\u0010\u0011\u001a\b\u0012\u0004\u0012\u00020\u00120\b\u00a2\u0006\b\n\u0000\u001a\u0004\b\u0013\u0010\f\u00a8\u0006\u0014"}, d2 = {"Leu/villacristina/lechugapro/ui/comercializacion/cliente/ClienteListaViewModel;", "Landroidx/lifecycle/ViewModel;", "repository", "Leu/villacristina/lechugapro/data/ClienteRepository;", "ingresoRepository", "Leu/villacristina/lechugapro/data/IngresoRepository;", "(Leu/villacristina/lechugapro/data/ClienteRepository;Leu/villacristina/lechugapro/data/IngresoRepository;)V", "clientesConDeuda", "Lkotlinx/coroutines/flow/StateFlow;", "", "", "getClientesConDeuda", "()Lkotlinx/coroutines/flow/StateFlow;", "todosLosClientes", "", "Leu/villacristina/lechugapro/data/Cliente;", "getTodosLosClientes", "totalIngresos", "", "getTotalIngresos", "app_debug"})
public final class ClienteListaViewModel extends androidx.lifecycle.ViewModel {
    @org.jetbrains.annotations.NotNull()
    private final kotlinx.coroutines.flow.StateFlow<java.util.List<eu.villacristina.lechugapro.data.Cliente>> todosLosClientes = null;
    @org.jetbrains.annotations.NotNull()
    private final kotlinx.coroutines.flow.StateFlow<java.lang.Double> totalIngresos = null;
    @org.jetbrains.annotations.NotNull()
    private final kotlinx.coroutines.flow.StateFlow<java.util.Set<java.lang.Long>> clientesConDeuda = null;
    
    public ClienteListaViewModel(@org.jetbrains.annotations.NotNull()
    eu.villacristina.lechugapro.data.ClienteRepository repository, @org.jetbrains.annotations.NotNull()
    eu.villacristina.lechugapro.data.IngresoRepository ingresoRepository) {
        super();
    }
    
    @org.jetbrains.annotations.NotNull()
    public final kotlinx.coroutines.flow.StateFlow<java.util.List<eu.villacristina.lechugapro.data.Cliente>> getTodosLosClientes() {
        return null;
    }
    
    @org.jetbrains.annotations.NotNull()
    public final kotlinx.coroutines.flow.StateFlow<java.lang.Double> getTotalIngresos() {
        return null;
    }
    
    @org.jetbrains.annotations.NotNull()
    public final kotlinx.coroutines.flow.StateFlow<java.util.Set<java.lang.Long>> getClientesConDeuda() {
        return null;
    }
}

File: app/src/main/java/eu/villacristina/lechugapro/ui/comercializacion/cliente/ClienteListaViewModel.kt
package eu.villacristina.lechugapro.ui.comercializacion.cliente

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider
import androidx.lifecycle.viewModelScope
import eu.villacristina.lechugapro.data.Cliente
import eu.villacristina.lechugapro.data.ClienteRepository
import eu.villacristina.lechugapro.data.IngresoRepository
import kotlinx.coroutines.flow.map
import kotlinx.coroutines.flow.SharingStarted
import kotlinx.coroutines.flow.StateFlow
import kotlinx.coroutines.flow.stateIn

class ClienteListaViewModel(
    repository: ClienteRepository,
    ingresoRepository: IngresoRepository
) : ViewModel() {

    val todosLosClientes: StateFlow<List<Cliente>> = repository.todosLosClientes
        .stateIn(
            scope = viewModelScope,
            started = SharingStarted.WhileSubscribed(5000),
            initialValue = emptyList()
        )

    // Total de ingresos sumados de todos los clientes
    val totalIngresos: StateFlow<Double> = ingresoRepository.todosLosIngresos
        .map { lista -> lista.sumOf { it.importe } }
        .stateIn(
            scope = viewModelScope,
            started = SharingStarted.WhileSubscribed(5000),
            initialValue = 0.0
        )
}

class ClienteListaViewModelFactory(
    private val repository: ClienteRepository,
    private val ingresoRepository: IngresoRepository
) : ViewModelProvider.Factory {
    override fun <T : ViewModel> create(modelClass: Class<T>): T {
        if (modelClass.isAssignableFrom(ClienteListaViewModel::class.java)) {
            @Suppress("UNCHECKED_CAST")
            return ClienteListaViewModel(repository, ingresoRepository) as T
        }
        throw IllegalArgumentException("Unknown ViewModel class")
    }
}
